### PR TITLE
feat: refactor config and files storing (APP-154)

### PR DESCRIPTION
**Pull request by anatolykislov**
- JAR directory changed
- Configs and data files (e.g. longevity incremental update data, models) storing in the same directory where JAR located

### CODE REVIEW

File: src/main/kotlin/app/utils/FileHelper.kt [hunk 1]
```diff
@@ -3,12 +3,45 @@
...
+                                           missingDelimiterValue = "")
+    }
+
+    fun getJarPath(): Path {
+        val fullPathString = URLDecoder.decode(FileHelper::class.java
+            .protectionDomain.codeSource.location.toURI().path, "UTF-8")
+        val fullPath = Paths.get(fullPathString)
+        val root = fullPath.root
+        // Removing jar filename.
+        return root.resolve(fullPath.subpath(0, fullPath.nameCount - 1))
```
Comment (asurkov) [inline src/main/kotlin/app/utils/FileHelper.kt]: might be unexpected behavior for the user, if the user moved a jar file, and we put all our stuff next to it, @sergey-surkov is it ok?

Comment (anatolykislov) [reply src/main/kotlin/app/utils/FileHelper.kt]: Why would the user want to do it? We use bash script that points to a specific location of the JAR to run sourcerer, all will be broken anyway.

File: src/main/kotlin/app/utils/FileHelper.kt [hunk 2]
```diff
@@ -3,12 +3,45 @@
 
 package app.utils
 
+import java.io.File
+import java.net.URLDecoder
+import java.nio.file.Files
 import java.nio.file.Paths
+import java.nio.file.Path
 
 object FileHelper {
```
[the remark this answers is not in the document]
Comment (anatolykislov) [reply src/main/kotlin/app/utils/FileHelper.kt]: Done.

File: src/main/kotlin/app/utils/FileHelper.kt [hunk 3]
```diff
@@ -3,12 +3,45 @@
...
+            Files.createDirectories(path)
+        }
+        return path.resolve(name)
+    }
+
+    fun getFile(name: String, vararg parts: String): File {
+        return getPath(name, *parts).toFile()
+    }
+
+    fun isNotExists(name:String, vararg parts: String): Boolean {
```
[the remark this answers is not in the document]
Comment (anatolykislov) [reply src/main/kotlin/app/utils/FileHelper.kt]: Done.

**Pull request merged**